shavee-bin/src/main.rs

```diff
@@ -1,90 +1,54 @@
 mod args;
 
-use args::Sargs;
+use args::{Sargs, Mode, Umode};
 use base64::encode_config;
 use shavee_lib::logic::{create_zfs_file, create_zfs_yubi, unlock_zfs_pass};
 use shavee_lib::logic::{print_mode_file, print_mode_yubi, unlock_zfs_file, unlock_zfs_yubi};
 use shavee_lib::password::hash_argon2;
 use shavee_lib::zfs::*;
+use std::error::Error;
 use std::process::exit;
 
+// main() collect the arguments from command line, pass them to run() and print any error 
+// message upon exiting the program
 fn main() {
     let args = Sargs::new();
-    let pass = rpassword::prompt_password_stderr("Dataset Password: ");
-    let pass = match pass {
-        Ok(pass) => pass,
+
+    // Only main() will terminate the executable with proper error message and code
+    exit(match run(args){
+        Ok(()) => 0,
         Err(error) => {
-            eprintln!("Error: Failed to read Password");
             eprintln!("Error: {}", error);
-            exit(1)
+            1
         }
-    };
+    });
+}
+
+
+fn run(args: Sargs) -> Result<(), Box<dyn Error>> {
+
+    let pass = rpassword::prompt_password_stderr("Dataset Password: ")
+        .map_err(|e| e.to_string())?;
```

Comment:
Just like `map` that operates on `Ok(i)`, `map_err` operates on `Err(e)`. 

Here we need to convert the error to `String` which can be converted to `Box<dyn Error>`